### PR TITLE
[MAINTENANCE] Test remaining expectations

### PR DESCRIPTION
**Pull request by tyler-hoffman**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, visit our [community resources](https://docs.greatexpectations.io/docs/core/introduction/community_resources#contribute-code-or-documentation).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/expectations/core/expect_table_columns_to_match_set.py [hunk 1]
```diff
@@ -170,7 +170,9 @@ class ExpectTableColumnsToMatchSet(BatchExpectation):
     column_set: Union[list, set, SuiteParameterDict, None] = pydantic.Field(
         description=COLUMN_SET_DESCRIPTION
     )
-    exact_match: Union[bool, None] = pydantic.Field(description=EXACT_MATCH_DESCRIPTION)
+    exact_match: Union[bool, None] = pydantic.Field(
+        default=None, description=EXACT_MATCH_DESCRIPTION
```
Comment (tyler-hoffman) [inline great_expectations/expectations/core/expect_table_columns_to_match_set.py:174]: This is to make mypy happy, and keeps behavior the same as before. It's actually [a bug to not default to True](https://greatexpectations.io/expectations/expect_table_columns_to_match_set), though. I want to fix that in a separate PR so we can separate the [MAINTENANCE] vs [BUGFIX] labels.

File: tests/integration/data_sources_and_expectations/expectations/test_expect_table_column_count_to_be_between.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+) -> None:
+    result = batch_for_datasource.validate(expectation)
+    assert not result.success
+
+
+@pytest.mark.xfail(reason="Fails at validation, but should fail when instantiating")
+@pytest.mark.unit
+def test_valid_range() -> None:
+    with pytest.raises(ValueError):
+        gxe.ExpectTableColumnCountToBeBetween(min_value=5, max_value=4)
```
Comment (cdkini) [inline tests/integration/data_sources_and_expectations/expectations/test_expect_table_column_count_to_be_between.py]: For these should we assert against the message at all? This is fine by me tbh

Comment (tyler-hoffman) [reply tests/integration/data_sources_and_expectations/expectations/test_expect_table_column_count_to_be_between.py]: Yeah, seems legit. Updated.

File: tests/integration/data_sources_and_expectations/expectations/test_expect_compound_column_values_to_be_unique.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            id="mostly_threshold_not_met",
+        ),
+    ],
+)
+@parameterize_batch_for_data_sources(data_source_configs=JUST_PANDAS_DATA_SOURCES, data=DATA)
+def test_failure(
+    batch_for_datasource: Batch, expectation: gxe.ExpectCompoundColumnsToBeUnique
+) -> None:
+    result = batch_for_datasource.validate(expectation)
+    assert not result.success
```
Comment (cdkini) [inline tests/integration/data_sources_and_expectations/expectations/test_expect_compound_column_values_to_be_unique.py:77]: Anytime we fail should we be checking the unexpected values list or anything else to confirm the reason behind the failure? Maybe overkill here

File: tests/integration/data_sources_and_expectations/expectations/test_expect_table_column_count_to_be_between.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+@parameterize_batch_for_data_sources(data_source_configs=JUST_PANDAS_DATA_SOURCES, data=DATA)
+def test_failure(
+    batch_for_datasource: Batch, expectation: gxe.ExpectTableColumnCountToBeBetween
+) -> None:
+    result = batch_for_datasource.validate(expectation)
+    assert not result.success
+
+
+@pytest.mark.xfail(reason="Fails at validation, but should fail when instantiating")
+@pytest.mark.unit
```
Comment (cdkini) [inline tests/integration/data_sources_and_expectations/expectations/test_expect_table_column_count_to_be_between.py:92]: These xfails will have corresponding tickets right?

File: tests/integration/data_sources_and_expectations/expectations/test_expect_table_column_count_to_equal.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import pandas as pd
+
+import great_expectations.expectations as gxe
+from great_expectations.datasource.fluent.interfaces import Batch
+from tests.integration.conftest import parameterize_batch_for_data_sources
+from tests.integration.data_sources_and_expectations.test_canonical_expectations import (
+    ALL_DATA_SOURCES,
+    JUST_PANDAS_DATA_SOURCES,
+)
```
Comment (cdkini) [inline tests/integration/data_sources_and_expectations/expectations/test_expect_table_column_count_to_equal.py:9]: Nit - not super important but could we have one file that defines these groupings? Would be good to have a single source of truth between test files?

**Review by cdkini — APPROVED** [review]

File: tests/integration/data_sources_and_expectations/expectations/test_expect_compound_column_values_to_be_unique.py (hunk 3 above)
Comment (tyler-hoffman) [reply tests/integration/data_sources_and_expectations/expectations/test_expect_compound_column_values_to_be_unique.py:77]: I found so many discrepancies that I don't think it's worth it. We could always go back and probably test categories of expectations against different data sources, but I think it's out of scope here.

File: tests/integration/data_sources_and_expectations/expectations/test_expect_table_column_count_to_be_between.py (hunk 4 above)
Comment (tyler-hoffman) [reply tests/integration/data_sources_and_expectations/expectations/test_expect_table_column_count_to_be_between.py:92]: They do!

File: tests/integration/data_sources_and_expectations/expectations/test_expect_table_column_count_to_equal.py (hunk 5 above)
Comment (tyler-hoffman) [reply tests/integration/data_sources_and_expectations/expectations/test_expect_table_column_count_to_equal.py:9]: Yeah, I'll circle back and do that in a PR tomorrow! It's been on my mind too

**Pull request merged**